src/extension.js

```diff
@@ -6,51 +6,71 @@
+import { createSyncFn } from 'synckit';
+import requireRelative from 'require-relative';
 
-import format from './formatter';
-import { isFilePathMatchedByEslintIgnore, isFilePathMatchedByPrettierIgnore } from './ignore';
-
-const path = require('path');
+import isFilePathMatchedByIgnore from './ignore';
 
 let outputChannel;
 
-async function formatter(document) {
+/**
+ * Resolves the path of a module relative to a file path.
+ * @param {string} filePath - The file path to resolve the module path relative to.
+ * @param {string} moduleName - The name of the module to resolve.
+ * @returns {string} - The resolved module path.
+ */
+function getModulePath(filePath, moduleName) {
+  try {
+    return requireRelative.resolve(moduleName, filePath);
+  } catch (error) {
+    return require.resolve(moduleName);
+  }
+}
+
+/**
+ * Formats the given document using 'prettier-eslint' package.
+ * If the document is ignored by either Prettier or ESLint,
+ * logs a message to the console and does nothing.
+ * @param {TextDocument} document - The document to format.
+ * @returns {TextEdit[]} - Formatted text edits.
+ */
+function formatter(document) {
   const documentPath = path.dirname(document.fileName);
   const workspaceDir = workspace?.workspaceFolders.find((w) => documentPath.startsWith(w.uri.path))
     ?.uri.path;
+
   try {
-    if (
-      isFilePathMatchedByEslintIgnore(document.fileName, workspaceDir)
-      || isFilePathMatchedByPrettierIgnore(document.fileName, workspaceDir)
-      // eslint-disable-next-line no-empty
-    ) {
-      console.log('File ignored.');
-      outputChannel.appendLine('File ignored. Matches entry in .eslintignore or .prettierignore');
-    } else {
-      const startLine = document.lineAt(0);
-      const endLine = document.lineAt(document.lineCount - 1);
-      const range = new Range(startLine.range.start, endLine.range.end);
+    if (isFilePathMatchedByIgnore(document.fileName, workspaceDir, '.eslintignore')) {
+      outputChannel.appendLine(`File ${document.fileName} is ignored by ESLint.`);
+      return [];
+    }
 
-      const text = document.getText(range);
-      const extensionConfig = workspace?.getConfiguration('vs-code-prettier-eslint');
-      const formatted = await format({
-        text,
-        filePath: document.fileName,
-        extensionConfig,
-      });
-      return [TextEdit.replace(range, formatted)];
+    if (isFilePathMatchedByIgnore(document.fileName, workspaceDir, '.prettierignore')) {
+      outputChannel.appendLine(`File ${document.fileName} is ignored by Prettier.`);
+      return [];
     }
+
+    const startLine = document.lineAt(0);
+    const endLine = document.lineAt(document.lineCount - 1);
+    const range = new Range(startLine.range.start, endLine.range.end);
+    const text = document.getText(range);
+    const prettierLast = workspace.getConfiguration('vs-code-prettier-eslint').get('prettierLast');
+
+    const formatText = createSyncFn(require.resolve('./worker'));
```

Comment:
https://github.com/idahogurl/vs-code-prettier-eslint/issues/171#issuecomment-1848082004

Move it outside to preload worker before first run.